core/src/main/java/io/wcm/testing/mock/aem/dam/MockGraniteAssetWrapper.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * #%L
+ * wcm.io
+ * %%
+ * Copyright (C) 2024 wcm.io
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package io.wcm.testing.mock.aem.dam;
+
+import java.io.InputStream;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import javax.jcr.Binary;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceWrapper;
+
+import com.adobe.granite.asset.api.Asset;
+import com.adobe.granite.asset.api.AssetMetadata;
+import com.adobe.granite.asset.api.AssetRelation;
+import com.adobe.granite.asset.api.Rendition;
+
+/**
+ * Mock implementation of Adobe Granite {@link Asset}. This is done by wrapping a {@link MockAsset}
+ */
+@SuppressWarnings("null")
+public class MockGraniteAssetWrapper extends ResourceWrapper implements Asset {
+
+  private final com.day.cq.dam.api.Asset asset;
+
+  MockGraniteAssetWrapper(com.day.cq.dam.api.Asset asset) {
+    super(asset.adaptTo(Resource.class));
+    this.asset = asset;
+  }
+
+  @Override
+  public <AdapterType> AdapterType adaptTo(Class<AdapterType> type) {
+    //to be able to adapt to CQ asset
+    if (type == com.day.cq.dam.api.Asset.class) {
+      return type.cast(asset);
+    }
+    return asset.adaptTo(type);
+  }
+
+  @Override
+  public Rendition getRendition(String s) {
+    return (Rendition)asset.getRendition(s);
+  }
+
+  @Override
+  public Iterator<? extends Rendition> listRenditions() {
+    List<Rendition> graniteRenditions = new LinkedList<>();
+    Iterator<com.day.cq.dam.api.Rendition> renditions = asset.listRenditions();
+    while (renditions.hasNext()) {
+      graniteRenditions.add((Rendition)renditions.next());
+    }
+    return graniteRenditions.iterator();
+  }
+
+  @Override
+  public String getIdentifier() {
+    return asset.getID();
+  }
+
+  @Override
+  public Rendition setRendition(String s, InputStream inputStream, Map<String, Object> map) {
+    if (map.size() == 1) {
```

Comment:
i do not understand this special logic with map size = 1, and it's also not tested with maps with > 1 entries (and i think using such a map would lead to an unsupported method).

according to https://developer.adobe.com/experience-manager/reference-materials/cloud-service/javadoc/com/adobe/granite/asset/api/Asset.html#setRendition(java.lang.String,java.io.InputStream,java.util.Map) the map is passed through to a RenditionHandler, and that interface defines only two supported properties: https://developer.adobe.com/experience-manager/reference-materials/cloud-service/javadoc/com/adobe/granite/asset/api/RenditionHandler.html

we do not have to mock that handler, but i think for the mock the best solution is to look for an entry with PROPERTY_RENDITION_MIME_TYPE key and ignore the rest of the map. it's not necessarily the first entry.

Reply:
i will quickly apply the changes here and update the test cases

Reply:
It was a way of being able to pass only thr mime type but your solution is more elegant.